src/common/ErrorMessage.jsx

```diff
@@ -1,15 +1,18 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-import exclamationIcon from '../fixtures/icons';
-
 const ErrorMessage = (props) => {
-  const { message, retry, param } = props;
+  const {
+    message,
+    retry,
+    param,
+    icon,
+  } = props;
   return (
     <div className='error-message'>
-      <img src={exclamationIcon} alt='error-icon' />
+      {icon && <img src={icon} alt='error-icon' />}
```

Comment:
I am always a fan of evaluating this html before the render i.:

`let` iconHtml = "";
if(icon){
iconHtml = <img src={icon} alt='error-icon' />
}
`
And then place that`iconHtml` in the render function. It feels more readable in the long run. Consider doing the same for line 15 for retry.